stl/inc/filesystem

```diff
@@ -4337,6 +4337,20 @@ namespace filesystem {
         return _STD filesystem::copy(_From, _To, copy_options::none);
     }
 } // namespace filesystem
+
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    template <>
+    inline constexpr bool enable_borrowed_range<filesystem::directory_iterator> = true;
+    template <>
+    inline constexpr bool enable_borrowed_range<filesystem::recursive_directory_iterator> = true;
+    template <>
+    inline constexpr bool enable_view<filesystem::directory_iterator> = true;
+    template <>
+    inline constexpr bool enable_view<filesystem::recursive_directory_iterator> = true;
+} // namespace ranges
+#endif // __cpp_lib_concepts
+

```

Comment:
Hint to other reviewers: our `end` overloads for `directory_iterator` and `recursive_directory_iterator` already take their argument by value as in the proposed resolution of LWG-3480. In fact they always have, all the way back to MSVC-PR-108103! I'll have to thank Gor for realizing that WG21 would want to change these 3.5 years later =)

[No change requested.]